presto-hive/src/test/java/com/facebook/presto/hive/AbstractTestHiveClient.java

```diff
@@ -1278,8 +1280,8 @@ public void testMismatchSchemaTable()
         boolean pushdownFilterEnabled = getHiveClientConfig().isPushdownFilterEnabled();
 
         for (HiveStorageFormat storageFormat : createTableFormats) {
-            // TODO: fix coercion for JSON
-            if (storageFormat == JSON) {
+            // TODO: fix coercion for JSON or PAGEFILE
+            if (storageFormat == JSON || storageFormat == PAGEFILE) {
```

Comment:
It should be pretty simple to implement schema evolution. However since it is not needed I aggree we shouldn't be doing this. The only recommendation I have is to add a specific error message (currently we are failing with ArrayOutOfBound)

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: 13
	at com.facebook.presto.spi.Page.getBlock(Page.java:103)
	at com.facebook.presto.hive.pagefile.PageFilePageSource.getNextPage(PageFilePageSource.java:99)
	at com.facebook.presto.hive.HivePageSource.getNextPage(HivePageSource.java:126)
	... 29 more

```

Reply:
Index check is added.